v2/oauth/multiple-frontends-with-separate-backends.mdx

```diff
@@ -0,0 +1,427 @@
+
+## 2. Initialize the provider recipe 
+
+In your [**Authentication Service**](/docs/oauth/introduction#authentication-service) you will need to initialize the **OAuth2Provider** recipe.
+The recipe will expose the endpoints needed for enabling the [**OAuth 2.0**](/docs/oauth/introduction#oauth-20) flow.
+
+
+<BackendSDKTabs>
+<TabItem value="nodejs">
+
+```typescript
+import supertokens from "supertokens-node";
+import OAuth2Provider from"supertokens-node/recipe/oauth2provider";
+
+supertokens.init({
+    framework: "express",
+    supertokens: {
+        connectionURI: "<YOUR_CONNECTION_URI>",
+        apiKey: "<YOUR_API_KEY>",
+    },
+    appInfo: {
+        appName: "",
+        apiDomain: "<YOUR_API_DOMAIN>",
+        websiteDomain: "<YOUR_WEBSITE_DOMAIN>",
+        apiBasePath: "/auth",
+        websiteBasePath: "/auth"
+    },
+    recipeList: [
+        // This is where you initialize the OAuth2 recipe
+        OAuth2Provider.init(),
+    ]
+});
+```
+
+</TabItem>
+
+<TabItem value="go">
+
+:::caution
+
+At the moment we do not have support creating OAuth2 providers in the Go SDK.
+
+:::
+
+</TabItem>
+
+<TabItem value="python">
+
+:::caution
+
+At the moment we do not have support creating OAuth2 providers in the Python SDK.
+
+:::
+
+</TabItem>
+
+</BackendSDKTabs>
+
+## 3. Set Up Session Handling in Each Application
+
+You will have to setup custom session handling logic in each of your `backend applications`.
+
+<BackendSDKTabs>
+
+<TabItem value="nodejs">
+
+You can use a generic OIDC or OAuth2 library. 
+For `NodeJS` we recommend using either [passport-oauth2](https://www.passportjs.org/packages/passport-oauth2/) or [passport-openidconnect](https://www.passportjs.org/packages/passport-openidconnect/).
+
+#### Example using `passport-oauth2`:
+
+```typescript
+import express, { Request, Response } from 'express';
+import passport from 'passport';
+import { Strategy as OAuth2Strategy } from 'passport-oauth2';
+import session from 'express-session';
+
+const AUTHORIZATION_SERVER_URL = 'https://auth.example.com';
+// The client ID returned from the first step
+const CLIENT_ID = 'your-client-id';
+// The client secret returned from the first step
+const CLIENT_SECRET = 'your-client-secret';
+// On of the callback URLs specified in the create client payload 
+const CALLBACK_URL = 'https://your-app.com/auth/callback';
+
+const app = express();
+
+app.use(session({
+  secret: 'your-session-secret',
+  resave: false,
+  saveUninitialized: false
+}));
+
+app.use(passport.initialize());
+app.use(passport.session());
+
+passport.serializeUser((user, done) => {
+  done(null, user);
+});
+
+passport.deserializeUser((obj, done) => {
+  done(null, obj);
+});
+
+passport.use(new OAuth2Strategy({
+  authorizationURL: `${AUTHORIZATION_SERVER_URL}/authorize`,
+  tokenURL: `${AUTHORIZATION_SERVER_URL}/token`,
```

Comment:
are these actually the correct paths? What should the user use for AUTHORIZATION_SERVER_URL?

Reply:
It should be something like 
- `{AUTH_API_DOMAIN}/{AUTH_API_BASE_PATH}/oauth/auth`
- `{AUTH_API_DOMAIN}/{AUTH_API_BASE_PATH}/oauth/token`

Reply:
resolved